android/app/src/main/java/org/pathcheck/covidsafepaths/exposurenotifications/ExposureNotificationClientWrapper.java

```diff
@@ -89,12 +104,63 @@ public void showPermissionDialog(ReactContext reactContext, ApiException apiExce
     return exposureNotificationClient.getTemporaryExposureKeyHistory();
   }
 
-  public Task<Void> provideDiagnosisKeys(List<File> files) {
-    return exposureNotificationClient.provideDiagnosisKeys(files);
+  public ListenableFuture<Void> provideDiagnosisKeys(List<File> files) {
+    // Update the configuration each time we download keys
+    return FluentFuture.from(config.refresh())
+        .catching(Exception.class, exception -> {
+          Log.d(TAG, "Config refresh failed: " + exception);
+          // Ignore the error, use the default config
+          return null;
+        }, AppExecutors.getLightweightExecutor())
+        .transformAsync((done) -> setDiagnosisKeysDataMapping(), AppExecutors.getBackgroundExecutor())
+        .catching(Exception.class, exception -> {
+          Log.d(TAG, "setDiagnosisKeysDataMapping failed: " + exception);
+          // Ignore the error, this method throws an error if it's called multiple times
+          return null;
+        }, AppExecutors.getLightweightExecutor())
+        .transformAsync((done) -> provideDiagnosisKeysFuture(files), AppExecutors.getBackgroundExecutor());
+  }
+
+  private ListenableFuture<Void> provideDiagnosisKeysFuture(List<File> files) {
+    Log.d(TAG, "provideDiagnosisKeys called with " + files.size() + " files");
+    return TaskToFutureAdapter.getFutureWithTimeout(
+        exposureNotificationClient.provideDiagnosisKeys(files),
+        PROVIDE_KEYS_TIMEOUT.toMillis(),
+        TimeUnit.MILLISECONDS,
+        AppExecutors.getScheduledExecutor());
+  }
+
+  private ListenableFuture<Void> setDiagnosisKeysDataMapping() {
+    Log.d(TAG, "setDiagnosisKeysDataMapping called with config " + config.getDiagnosisKeysDataMapping());
+    return TaskToFutureAdapter.getFutureWithTimeout(
+        exposureNotificationClient.setDiagnosisKeysDataMapping(config.getDiagnosisKeysDataMapping()),
+        SET_DIAGNOSIS_KEY_DATA_MAPPING_TIMEOUT.toMillis(),
+        TimeUnit.MILLISECONDS,
+        AppExecutors.getScheduledExecutor());
   }
 
-  public Task<List<ExposureWindow>> getExposureWindows() {
-    return exposureNotificationClient.getExposureWindows();
+  public ListenableFuture<List<DailySummary>> getDailySummaries() {
+    return FluentFuture.from(
+        TaskToFutureAdapter.getFutureWithTimeout(
+            exposureNotificationClient.getDailySummaries(config.getDailySummariesConfig()),
+            GET_DAILY_SUMMARIES_TIMEOUT.toMillis(),
+            TimeUnit.MILLISECONDS,
+            AppExecutors.getScheduledExecutor())
+    ).transform(summaries -> {
+      if (summaries == null) {
+        return null;
+      }
+
+      List<DailySummary> filteredSummaries = new ArrayList<>();
+      for (DailySummary summary : summaries) {
+        if (summary.getSummaryData().getWeightedDurationSum() >= config.getTriggerThresholdWeightedDuration()) {
```

Comment:
Ugh, I made terrible life choices when I named/set this!
trigger threshold weighted duration is set in minutes.

I *was* fairly certain that the weighted duration is calculated as seconds, but after checking I'm not sure, the API is a little unclear.
https://developers.google.com/android/exposure-notifications/exposure-notifications-api#dailysummariesconfig
I would suggest you check this.

If it's minutes I guess we're good, although it's confusing since in a lot of place the API uses seconds.

Reply:
They return some values in minutes and some other in seconds, so I'm not sure either.
The documentation is not clear, so I'll test it to check what it returns

Reply:
I checked and it returns a value in seconds, I got `7020` and it can't be in minutes because those are daily summaries.
I'll update this PR

Reply:
Great, thanks!